### PR TITLE
Fixes for using default ciphers

### PR DESCRIPTION
**Pull request by kaleb-himes**
Thanks to Johan Kleuskens for noticing this typo, should be DHE_RSA_WITH, not DHE_WITH_RSA

BUILD_TLS_DHE_WITH_RSA_CAMELLIA_256_CBC_SHA (typo)
BUILD_TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA (fix)

### CODE REVIEW

**Comment by ejohnstown** [comment]
I disagree. This typo is in InitSuites(), which affects the cipher suite list used in the hello messages. The automated test is using wolfSSL_CTX_set_cipher_list() which replaces the suite list with anything.

**Comment by toddouska** [comment]
Then why does it work when you force the cipher suite on one side of the connection but not the other?  Because the same cipher suite is valid a few cipher suites later in InitSuites().  This typo is redundant (that is the whole cipher suite is redundant here) and not necessary.

**Comment by dgarske** [comment]
Hey Guys,

Update on my findings.

1. The typo fix is correct.
2. The reason for the failure not being detected is because when wolfSSL_CTX_set_cipher_list is called the "suites->setSuites" flag is set, which causes the "InitSuites" to exit early.
3. As for Todd's comment about redundant, that is incorrect. There is only one section for "BUILD_TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA". There is another one for SHA256 "BUILD_TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA256" right below, which might be the confusion.
4. This also turned up other failures that I'm correcting. For example the "DHE-RSA-AES128-SHA256" and "DHE-RSA-CHACHA20-POLY1305" tests with old TLS fail. The SetCipherList function looks up using name only and sets the suite. The InitSuites function uses additional information such as tls1_2 flags. This causes these tests to fail with the new code because InitSuites says they are TLS 1.2 only. I'm working with Jacob to identify which is valid or not and either correct test.conf or the InitSuites.

More Info:
Using our example server / client manually:
```
./examples/server/server -v 1 -l DHE-RSA-CAMELLIA256-SHA
./examples/client/client  -v 1 -l DHE-RSA-CAMELLIA256-SHA
```
The server finds the cipher suite in the `cipher_names` array based on name and is done at the WOLFSSL_CTX level. Then when the WOLFSSL object is created the `InitSuites` is called, but because `suites->setSuites` is set the section of code with the typo `BUILD_TLS_DHE_WITH_RSA_CAMELLIA_256_CBC_SHA` isn't hit.

The client example has the same issue. Since the cipher list has been set the the InitSuites doesn't get to the typo line.

The fix for this test is to have the server not set a cipher list and have the client try the cipher suite. This successfully causes a failure.

I should have all fixes pushed shortly.

File: src/internal.c [hunk 1]
```diff
@@ -10315,7 +10318,7 @@ static int BuildCertHashes(WOLFSSL* ssl, Hashes* hashes)
             #endif
         }
     }
-#if ! defined( NO_OLD_TLS )
+#if !defined(NO_OLD_TLS)
     else {
         BuildMD5_CertVerify(ssl, hashes->md5);
         BuildSHA_CertVerify(ssl, hashes->sha);
```
Comment (toddouska) [inline src/internal.c]: In the WOLFSSL_ALLOW_TLS_SHA1 case with TLS 1.2  and NO_OLD_TLS shouldn't the SHA1 hash be calculated here?

File: src/internal.c [hunk 2]
```diff
@@ -3573,6 +3573,7 @@ int InitSSL(WOLFSSL* ssl, WOLFSSL_CTX* ctx)
         WOLFSSL_MSG("HS_Hashes Memory error");
         return MEMORY_E;
     }
+    XMEMSET(ssl->hsHashes, 0, sizeof(HS_Hashes));
```
Comment (toddouska) [inline src/internal.c]: This looks like the wrong place to memset.  Shouldn't it be done right after allocation?

Comment (dgarske) [inline src/internal.c]: The allocation is done above at line 3570. This looks like the correct place to me...

**Comment by dgarske** [comment]
Just pushed a rebase to resolve the merge conflicts in the client/server examples.

File: src/internal.c (hunk 2 above)
Comment (toddouska) [inline src/internal.c]: You're right, I thought this was in InitSuites.  Thanks.

**Pull request merged**